### DISCUSSION TITLE
Getting weird Content-Type Headers in responses from `vercel dev`

### DISCUSSION THREAD

**Discussion by OscarSaharoy** [vercel/community#1577] [Help] [answered]
### Summary

I think this is a bug, I'm not sure what is happening but I think it started when I updated the vercel CLI and is now making `vercel dev` unusable

bug repro:

```
git clone git@github.com:OscarSaharoy/vercel-dev-test.git
cd vercel-dev-test
vercel dev
```
Then go to [http://localhost:3000](http://localhost:3000) in firefox and see the error in the console, check the Content-Type of the response for the request to http://localhost:3000/css/test.css

![image](https://user-images.githubusercontent.com/29902113/220139787-08da7050-cff4-4637-81ee-9560828c74bb.png)
![image](https://user-images.githubusercontent.com/29902113/220140032-3676f303-2ae5-4afa-944e-10fe1500346d.png)

Its the filename for some reason

### Example

https://github.com/OscarSaharoy/vercel-dev-test

**Comment by OscarSaharoy** [comment]
Think i found the problem: 


vercel/packages/cli/src/util/dev/mime-type.ts
```
import { contentType } from 'mime-types';
 
export default function getMimeType(fileName: string) {
   return contentType(fileName) || 'application/octet-stream';
}
```


vercel/packages/cli/node_modules/mime-types/index.js
```
 70 /**
 71  * Create a full Content-Type header given a MIME type or extension.
 72  *
 73  * @param {string} str
 74  * @return {boolean|string}
 75  */
 76 
 77 function contentType (str) {
 78   // TODO: should this even be in this module?
 79   if (!str || typeof str !== 'string') {
 80     return false
 81   }
 82 
 83   var mime = str.indexOf('/') === -1
 84     ? exports.lookup(str)
 85     : str
 86 
 87   if (!mime) {
 88     return false
 89   }
 90 
 91   // TODO: use content-type or other module
 92   if (mime.indexOf('charset') === -1) {
 93     var charset = exports.charset(mime)
 94     if (charset) mime += '; charset=' + charset.toLowerCase()
 95   }
 96 
 97   return mime
 98 }
```

you can see the contentType function takes an extension but is being called with an entire filename. I will try to make a PR but if anyone can do it quickly that would be great :)

**Reply by PonomareVlad** [reply]
I also met this problem, after installing the current version of vercel cli

So, return to version `28.14.1` helped me 🫠

**Reply by EndangeredMassa** [reply]
Sorry for the trouble! We'll fix this soon. In the mean time, downgrading will keep you going.

**Reply by OscarSaharoy** [reply]
Thanks guys, I submitted this PR too: https://github.com/vercel/vercel/pull/9492
I'm sure its not up to standard but i think it shows where the problem is

**Comment by EndangeredMassa** [comment, marked as answer]
You can try `npm -g i https://vercel-9zxeo4fby.vercel.sh/tarballs/vercel.tgz` to see if https://github.com/vercel/vercel/pull/9498 fixes it. Or you can wait for the next release, which should be later today or tomorrow.

**Reply by EndangeredMassa** [reply]
This has been fixed in [vercel@28.16.5](https://github.com/vercel/vercel/releases/tag/vercel%4028.16.5). Let us know if you have any other issues!
